pkg/webservice/handlers.go

```diff
@@ -668,12 +669,17 @@ func getPartitionQueue(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	queueName := vars.ByName("queue")
-	queueErr := validateQueue(queueName)
+	escapedQueueName, err := url.QueryUnescape(queueName)
+	if err != nil {
+		buildJSONErrorResponse(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	queueErr := validateQueue(escapedQueueName)
```

Comment:
This approach results in fewer changes:
```
	queueName, err := url.QueryUnescape(vars.ByName("queue"))
	if err != nil {
		buildJSONErrorResponse(w, err.Error(), http.StatusBadRequest)
		return
	}
	queueErr := validateQueue(queueName)
```
We don't need the unescaped string.

Reply:
We need to throw "missing" kind of error before this unescaping step. In fact filed [jira](https://issues.apache.org/jira/browse/YUNIKORN-2712) to address the same earlier.

Reply:
I realized that the variable name is still incorrect. It should either be called "unescapedQueueName" or rename the original to "escapedQueueName". If the unescaped is used more frequently, I'd just leave it and would call the output of `vars.ByName("queue")` as "escapedQueueName".

Reply:
Maybe we can introduce a small helper for all params?
```go
func byName(vars httprouter.Params, name string) (string, error) {
	value := vars.ByName(name)
	unescape, err := url.QueryUnescape(value)
	if err != nil {
		return "", err
	}
	return unescape, nil
}

	queueName, err := byName(vars, "queue")
	if err != nil {
		buildJSONErrorResponse(w, err.Error(), http.StatusBadRequest)
		return
	}
```

https://issues.apache.org/jira/browse/YUNIKORN-2712 can leverage this helper to add `missing` check

Reply:
Sounds good to me.

Reply:
> I realized that the variable name is still incorrect. It should either be called "unescapedQueueName" or rename the original to "escapedQueueName". If the unescaped is used more frequently, I'd just leave it and would call the output of `vars.ByName("queue")` as "escapedQueueName".

Yes, it should be "unescapedQueueName". Made the changes.